openapi.yaml

```diff
@@ -23,901 +31,1078 @@ tags:
+    description: The number of the first record to be supplied in the segmented response message.
+    required: false
+    type: integer
+    format: int32
+    default: 0
+    minimum: 0
+    allowEmptyValue: false
+  sort:
+    name: sort
+    in: query
+    description: "Identifies the sort criteria to be used for the records in the response message.
+    Order is ascending unless prefixed with a '-'"
+    required: false
+    type: string
+    allowEmptyValue: false
+  filter:
+    name: filter
+    in: query
+    description: 'The filtering rules to be applied when identifying the records to be supplied in
+    the response message. The sort order should follow the [UNICODE, 16] standard.'
+    required: false
+    type: string
+    allowEmptyValue: false
+  fields:
+    name: fields
+    in: query
+    description: To identify the range of fields that should be supplied in the response message.
+    required: false
+    type: array
+    items:
+      type: string
+    collectionFormat: csv
+    allowEmptyValue: false
+  educationOfferingId:
+    name: id
+    in: path
+    description: ...tbd...
+    required: true
+    type: string
+  educationId:
+    name: id
+    in: path
+    description: ...tbd...
+    required: true
+    type: string
+  offeringAssociationId:
+    name: id
+    in: path
+    description: ...tbd...
+    required: true
+    type: string
+responses:
+  '400':
+    description: "Either an invalid selection field or invalid filter selection was supplied and
+      data filtering on the selection criteria was not possible i.e. 'invalid_selection_field' or
+      'invalid_filter_field' respectively."
+    schema:
+      $ref: '#/definitions/ErrorResult'
+  '401':
+    description: "The request was not correctly authorised i.e. 'unauthorisedrequest."
+    schema:
+      $ref: '#/definitions/ErrorResult'
+  '403':
+    description: "This is used to indicate that the server can be reached and process the request
+      but refuses to take any further action i.e. 'forbidden'."
+    schema:
+      $ref: '#/definitions/ErrorResult'
+  '404':
+    description: "Either the supplied identifier is unknown in the Service Provider and so the
+      object could not be changed or an invalid GUID has been supplied. The statement 'Unknown
+      Object' should also be presented."
+    schema:
+      $ref: '#/definitions/ErrorResult'
+  409Post:
+    description: The request body resource object's type was invalid or, if a client-generated id
+      was used, a resource already exists with this id
+    schema:
+      $ref: '#/definitions/ErrorResult'
+  409Patch:
+    description: The request body resource object had an invalid type, invalid id, or violated a
+      uniqueness constraint
+  '422':
+    description: This error condition may occur if a JSON request body contains well-formed (i.e.
+      syntactically correct), but semantically erroneous, JSON instructions.
+    schema:
+      $ref: '#/definitions/ErrorResult'
+  '429':
```

Comment:
I think that 429 is reserved for rate limiting, ie. the particular application is submitting too many requests. This is different from a 'server busy' response which implies that the server cannot keep up with the amount of requests from all applications at that instance, and so it should return an internal server error. I changed my 429 description to reflect this

Reply:
Oh interesting, and nice catch. In the case of a "server busy" error, should we return a `503 Service Unavailable` code instead? If I'm understanding it right, this would tell the client that the server can't respond to their request right now, but they should try again sometime later.

Reply:
Yes I would think 503 would be appropriate in that case. I think that what they wanted to include in the design is 429 for a rate limiting response, but if they truly meant "server busy" then 503 should be added.